### PR TITLE
Documentation on how to write unit tests using auto-generated mocked stubs.

### PR DESCRIPTION
**Pull request by MakMukhi**

### CODE REVIEW

File: doc/unit_testing.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# How to write unit tests for gRPC C++ client.
+
+tl;dr: [Example code](https://github.com/grpc/grpc/blob/master/test/cpp/end2end/mock_test.cc).
+
+To unit-test client-side logic via the synchronous API, gRPC provides a mocked Stub that can be programmed upon and easily incorporated in the test code.
```
Comment (yang-g) [inline doc/unit_testing.md]: mention this is based on googletest (googlemock)?

File: doc/unit_testing.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+
+```proto
+service EchoTestService {
+        rpc Echo(EchoRequest) returns (EchoResponse);
+        rpc BidiStream(stream EchoRequest) returns (stream EchoResponse);
+}
+```
+
+The code generate would look something like this:
```
Comment (yang-g) [inline doc/unit_testing.md]: s/generate/generated/

File: doc/unit_testing.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+```py
+grpc_proto_library(
+  name = "echo_proto",
+  srcs = ["echo.proto"],
+  generate_mock = True, 
+)
+```
+
+
+By adding such a flag now a header file `echo_mock.h` containing the mocked stub will also be generated.
```
Comment (yang-g) [inline doc/unit_testing.md]: echo_mock.grpc.pb.h

Comment (MakMukhi) [inline doc/unit_testing.md]: Thanks for the reviews. Updated the file.

**Pull request merged**